R/nfidd-stan-tools.r

```diff
@@ -0,0 +1,304 @@
+  stan_files <- list.files(
+    file.path(stan_path, "functions"),
+    pattern = "\\.stan$", full.names = TRUE,
+    recursive = TRUE
+  )
+  all_content <- character(0)
+
+  for (file in stan_files) {
+    content <- readLines(file)
+    if (is.null(functions)) {
+      all_content <- c(all_content, content)
+    } else {
+      for (func in functions) {
+        func_content <- .extract_stan_functions(
+          content,
+          names_only = FALSE,
+          functions = func
+        )
+        all_content <- c(all_content, func_content)
+      }
+    }
+  }
+
+  # Add version comment
+  version_comment <- paste(
+    "// Stan functions from nfidd version",
+    utils::packageVersion("nfidd")
+  )
+  all_content <- c(version_comment, all_content)
+
+  if (wrap_in_block) {
+    all_content <- c("functions {", all_content, "}")
+  }
+
+  result <- paste(all_content, collapse = "\n")
+
+  if (write_to_file) {
+    writeLines(result, output_file)
+    message("Stan functions written to: ", output_file, "\n")
+  }
+
+  return(result)
+}
+
+#' List Available Stan Models in NFIDD
+#'
+#' This function finds all available Stan models in the NFIDD package and
+#' returns their names without the .stan extension.
+#'
+#' @param stan_path Character string specifying the path to Stan files. Defaults
+#'   to the result of `nfidd_stan_path()`.
+#'
+#' @return A character vector of available Stan model names.
+#'
+#' @export
+#'
+#' @examples
+#' nfidd_list_stan_models()
+nfidd_stan_models <- function(
+    stan_path = nfidd::nfidd_stan_path()
+  ) {
+  stan_files <- list.files(
+    stan_path,
+    pattern = "\\.stan$", full.names = FALSE,
+    recursive = FALSE
+  )
+
+  # Remove .stan extension
+  model_names <- tools::file_path_sans_ext(stan_files)
+
+  return(model_names)
+}
+
+#' Create a CmdStanModel with NFIDD Stan functions
+#'
+#' This function creates a CmdStanModel object using a specified Stan model from
+#' the NFIDD package and optionally includes additional user-specified Stan
+#' files.
+#'
+#' @param model_name Character string specifying which Stan model to use.
+#' @param include_paths Character vector of paths to include for Stan
+#'  compilation. Defaults to the result of `nfidd_stan_path()`.
+#' @param ... Additional arguments passed to cmdstanr::cmdstan_model().
+#'
+#' @return A CmdStanModel object.
+#'
+#' @export
+#'
+#' @family modelhelpers
+#'
+#' @examplesIf requireNamespace("cmdstanr", quietly = TRUE)
+#' if (!is.null(cmdstanr::cmdstan_version(error_on_NA = FALSE))) {
+#'   model <- nfidd_cmdstan_model("renewal", compile = FALSE)
+#'   model
+#' }
+nfidd_cmdstan_model <- function(
+    model_name,
+    include_paths = nfidd::nfidd_stan_path(),
+    ...) {
+  if (!requireNamespace("cmdstanr", quietly = TRUE)) {
+    stop("Package 'cmdstanr' is required but not installed for this function.")
+  }
+
+  stan_model <- system.file(
+    "stan", paste0(model_name, ".stan"),
+    package = "nfidd"
+  )
```

Comment:
This doesn't work with `devtools::load_all()`. Perhaps not the end of the world but could be fixed by instead using 
```suggestion
  stan_model <- fs::path_package(
    "nfidd", "stan", paste0(model_name, ".stan")
  )
```

Reply:
Oh weird it works for me and I use it as a pattern everywhere?

Reply:
If changing we need to add to the description

Reply:
> If changing we need to add to the description

Yes. I'm OK to leave as is for now and handle via `devtools::install()` and reloading if/where needed.